analyzers/tests/SonarAnalyzer.Test/Rules/Utilities/AnalysisWarningAnalyzerTest.cs

```diff
@@ -37,37 +37,51 @@ public class AnalysisWarningAnalyzerTest
         File.Exists(expectedPath).Should().BeTrue();
-        File.ReadAllText(expectedPath).Should().Be("""[{"text": "Analysis using MsBuild 14 and 15 build tools is deprecated. Please update your pipeline to MsBuild 16 or higher."}]""");
+        File.ReadAllText(expectedPath).Should().Be("""[{"text": "The analysis using MsBuild 14 is no longer supported and the analysis with MsBuild 15 is deprecated. Please update your pipeline to MsBuild 16 or higher."}]""");
 
         // Lock file and run it for 2nd time
         using var lockedFile = new FileStream(expectedPath, FileMode.Open, FileAccess.Write, FileShare.None);
-        ExecuteAnalyzer(languageName, true, 1000).Should().Be(expectedPath, "path should be reused and analyzer should not fail");
+        ExecuteAnalyzer(languageName, true, 1000, 1001).Should().Be(expectedPath, "path should be reused and analyzer should not fail");
     }
 
     [DataTestMethod]
     [DataRow(LanguageNames.CSharp)]
     [DataRow(LanguageNames.VisualBasic)]
-    public void FileExceptions_AreIgnored(string languageName)
+    public void AnalysisWarning_MSBuild15UnsupportedScenario_GenerateWarning(string languageName)
+    {
+        var expectedPath = ExecuteAnalyzer(languageName, true, RoslynHelper.UnsupportedVersionVS2017, 1000); // Requiring too high Roslyn version => we're under unsupported scenario
+        File.Exists(expectedPath).Should().BeTrue();
+        File.ReadAllText(expectedPath).Should().Be("""[{"text": "The analysis using MsBuild 15 is deprecated. Please update your pipeline to MsBuild 16 or higher."}]""");
+
+        // Lock file and run it for 2nd time
+        using var lockedFile = new FileStream(expectedPath, FileMode.Open, FileAccess.Write, FileShare.None);
+        ExecuteAnalyzer(languageName, true, RoslynHelper.UnsupportedVersionVS2017, 1000).Should().Be(expectedPath, "path should be reused and analyzer should not fail");
+    }
+
+    [DataTestMethod]
+    [DataRow(LanguageNames.CSharp)]
+    [DataRow(LanguageNames.VisualBasic)]
+    public void AnalysisWarning_FileExceptions_AreIgnored(string languageName)
     {
         // This will not create the output directory, causing an exception in the File.WriteAllText(...)
-        var expectedPath = ExecuteAnalyzer(languageName, true, 1000, false);  // Requiring too high Roslyn version => we're under unsupported scenario
+        var expectedPath = ExecuteAnalyzer(languageName, true, 500, 1000, false);  // Requiring too high Roslyn version => we're under unsupported scenario
         File.Exists(expectedPath).Should().BeFalse();
     }
 
-    private string ExecuteAnalyzer(string languageName, bool isAnalyzerEnabled, int minimalSupportedRoslynVersion, bool createDirectory = true)
+    private string ExecuteAnalyzer(string languageName, bool isAnalyzerEnabled, int unsupportedVersionVS2017, int minimalSupportedRoslynVersion, bool createDirectory = true)
```

Comment:
One we have the dedicated test I think the `unsupportedVersionVS2017` parameter will no longer be needed. The file will be generated anyway if the `minimalSupportedRoslynVersion` does not match.